simplified-books-covers/src/main/java/org/nypl/simplified/books/covers/BookCoverGenerator.java

```diff
@@ -81,19 +77,20 @@ public BookCoverGenerator(
       final String author_maybe = params.get("author");
       final String author = author_maybe == null ? "" : NullCheck.notNull(author_maybe);
 
+      if (width == 0) {
+        width = (int) Math.round(height * .75);
```

Comment:
The width is now variable, so may not be known in advance. The code below expects the width to be greater than zero. Since we're generating covers we can give it any width we want. Calculate a reasonable width from the given height instead.